apps/src/lib/levelbuilder/rubrics/RubricsContainer.jsx

```diff
@@ -125,28 +195,42 @@ export default function RubricsContainer({
     })
       .then(response => response.json())
       .then(data => {
-        navigateToHref(data.redirectUrl);
+        if (!rubric) {
+          navigateToHref(data.redirectUrl);
+        }
       })
       .catch(err => {
         console.error('Error saving rubric:' + err);
       });
   };
 
-  // transforms keys from camelCase to snake_case for rails
-  // specifically created to do this for an array of objects
-  // with keys in camelCase
-  function transformKeys(startingList) {
-    const newList = [];
+  /**
+   * Transforms the keys of an object from camelCase to snake_case.
+   * @param {object} obj - The object with keys in camelCase format.
+   * @returns {object} - The object with keys in snake_case format.
+   */
+  function transformObjectKeys(obj) {
+    const newObj = {};
 
-    startingList.forEach(item => {
-      const newItem = {};
-      for (const [key, value] of Object.entries(item)) {
-        newItem[snakeCase(key)] = value;
+    for (const [key, value] of Object.entries(obj)) {
+      if (Array.isArray(value)) {
+        newObj[snakeCase(key)] = value.map(item => transformObjectKeys(item));
```

Comment:
recursive functions always make me feel weird. Can you add a comment that this is always expected to be no more than a single layer of nesting?